### DISCUSSION TITLE
Modify ERDDAP's default data license text to better align with current NOAA data licensing guidance

### DISCUSSION THREAD

**Discussion by mwengren** [ERDDAP/erddap#211] [Ideas]
The below ISO XML snippit represents the default dataset license text provided by ERDDAP: 

```
<gmd:resourceConstraints>
<gmd:MD_LegalConstraints>
<gmd:useLimitation>
<gco:CharacterString>The data may be used and redistributed for free but is not intended for legal use, since it may contain inaccuracies. Neither the data Contributor, ERD, NOAA, nor the United States Government, nor any of their employees or contractors, makes any warranty, express or implied, including warranties of merchantability and fitness for a particular purpose, or assumes any legal liability for the accuracy, completeness, or usefulness, of this information.</gco:CharacterString>
</gmd:useLimitation>
</gmd:MD_LegalConstraints>
</gmd:resourceConstraints>
``` 

### NOAA's current guidance for open data licensing for 'Internal NOAA Source Data': 

To implement in XML Metadata: When creating metadata for NOAA internal data, the following statement should be used to identify the use of the CC0-1.0 license: 

> "These data were produced by NOAA and are not subject to copyright protection in the United States. NOAA waives any potential copyright and related rights in these data worldwide through the Creative Commons Zero 1.0 Universal Public Domain Dedication (CC0-1.0)." 

A similar guidance statement for display/rendering on websites (same as the above but with the hyperlink rendered via markdown in the example below or via HTML in rendered pages):

> These data were produced by NOAA and are not subject to copyright protection in the United States. NOAA waives any potential copyright and related rights in these data worldwide through the [Creative Commons Zero 1.0 Universal Public Domain Dedication (CC0-1.0)](https://creativecommons.org/publicdomain/zero/1.0/).

Here are some (internal to NOAA only) documents that contain the above guidance as a reference:

https://docs.google.com/document/d/10PmqXVpYlCdpqe05bGBXkpZ6bKLdeyvZiOQ_6KAMdCs/edit
https://docs.google.com/document/d/1zFLEcJKjD0CJbM6IgikqfNQOJ5gXWeYyFAhTbqY40_g/edit

Neither of the above docs provide actual ISO XML examples of how the license statement should be encoded into XML, however, I would guess that replacing the ERDDAP default <gmd:useLimitation> text with the above text would do the trick.

I believe this guidance is part of the recently released NOAA Data Management Handbook (see Appendix E).

[NAO 212-15B - Data Management Handbook - released 2024 October 1.pdf](https://github.com/user-attachments/files/17282819/NAO.212-15B.-.Data.Management.Handbook.-.released.2024.October.1.pdf)

**Comment by rmendels** [comment]
@mwengren   - I still think you need the part including and  following "not intended for legal use ..."  - waiving copyright is important, but it is not the same as this.   Having potentially been involved in some legal proceedings that wanted to use data we provided  (how do you say Gulf of Mexico for example),  you get into some pretty sticky issues without such a statement.  Basically it is the difference between datasets that have been reviewed and given NOAA's official approval and other datasets.  In our case,  NOAA as an entity had not approved the datasets.  GC wanted us to remove the restriction,  we refused,  we would only say that we had every reason to believe that the data are correct and we take care to make it so,  but the dataset had not been approved by NOAA as an official,  approved, legal  dataset.  If the Data Handbook doesn't recognize that distinction,  then it should be modified.  It also protects you from a firm using your data,  suffer a loss that they consider is due to the data being off,  and suing for damages.  That statement basically says if they used the data it was at their own risk - waver of copyright does not cover that.

**Reply by mwengren** [reply]
Hi @rmendels.  You should have been involved with the Data Licensing task team that developed this recommendation!   I wasn't myself, but I understand this to be NOAA's official position on data licensing (for the use case 1 listed Appendix E in the handbook linked above).  

Use Case 1 applies to 'Internal NOAA Source Data' and is the only one listed in the handbook at present.  I noticed just now that it states also:

> To be added at a future date by the Data Licensing Task Team: [CASE 2: External source data with CC BY 4.0 International] [CASE 3: Sources that require a different type of license than CASE 1 or CASE 2]

I'm not sure the process to recommend revisions to the handbook, but I'd suspect it involves talking to your ACDO, who in turn could raise it at the DGC (Data Governance Cmte) level.  Or, maybe it's still possible to join the TT as it sounds like they have more work planned?

**Reply by rmendels** [reply]
@mwengren - my basic point is licensing can be a different issue than a liability waiver.  Licensing, say CC0 for example,  is who can use the data etc etc,  the waiver is a standard waiver for software - and it is the ERDDAP software,  not necessarily the data being served,  for which the licensing statement holds.  Each Admin is responsible for the data but not for ERDDAP itself.  Suppose ERDDAP makes an error in subsetting a dataset that a user feels has cost them money.  NOAA then could be sued  (well actually you can't waive negligence but that changes what a person suing has to prove).  Please go look at a lot of open source software,  such as R,  and look at their waiver.  In fact,  I think we initially used the R waiver as the statement,  but I can't remember if we modified it.  And as for being on that committee,  no thanks I spent way too much time on such committees,  I have a fair number of large door stops that I can use,   I am almost 75 and have a lot of other things to spend my time on.

**Comment by rmendels** [comment]
@mwengren that should be "waiver",  not "waver".  Plus we know of any number of commercial companies in a variety of areas that use our data,  so it is important they realize it is caveat emptor.  Not that we don't think our products are as correct as we can make them,  but that is different than when there is perceived legal harm and you are being sued.

**Comment by MathewBiddle** [comment]
This has been a thorn in my side for [some time](https://groups.google.com/g/erddap/c/NNeDnr2x0PU/m/edjZ0u0YBQAJ). As not all ERDDAP's are administered from NOAA offices, this NOAA language is not applicable to all ERDDAPs and shouldn't be the **default**. My opinion is that each ERDDAP administrator should be having a conversation with their community about what dataset licenses they should be using and at what granularity.

So, I recommend that the default license text in ERDDAP is removed (i.e. left blank). This removes assumptions about use constraints with the data and encourages administrators to make thoughtful decisions about what is contained in the field.

Maybe we find a place in the NOAA Data Handbook to put the recommended default license for NOAA Offices that @mwengren recommends, above?

**Reply by mwengren** [reply]
I think that's a good approach.  But how does this type of per-ERDDAP-instance override work from an administrator standpoint?

Does it require a) configuring license text on a per-dataset basis in `datasets.xml`, or b) is there a way to apply a custom license text string instance-wide, or, ideally c) a hierarchical approach with an instance-wide setting an an option to override per-dataset in `datasets.xml`?

If it's 'a', then I might still argue for adopting the NOAA-recommended licensing text as a default for ease-of-use and given ERDDAP's association with NOAA.   Or, better-yet, looking into a new feature to enable 'c' for more flexible customization.

**Reply by MathewBiddle** [reply]
I believe the answer is **c**. https://erddap.github.io/setupDatasetsXml.html#license

You can set an ERDDAP wide standard or on a per dataset.

**Reply by MathewBiddle** [reply]
I should state, that the current standard text can be updated. I just don't think folks do that. See https://github.com/ERDDAP/erddap/blob/1381745790a5b07aaf2fcf01173d147a5cda2e5d/src/main/resources/gov/noaa/pfel/erddap/util/messages.xml#L5395-L5403

**Reply by mwengren** [reply]
Maybe this could just be an ERDDAP documentation update with a reference to NOAA handbook's licensing section as an example?  The handbook is only on an internal site as far as I can tell, so that might not work unless it's made public somewhere.

Emphasizing that ERDDAP admins should pay attention to data licensing would be an improvement, for sure, in my opinion.
